samples/bluetooth/hci_spi/src/main.c

```diff
@@ -148,8 +148,8 @@ static void bt_tx_thread(void *p1, void *p2, void *p3)
 	uint8_t header_slave[5] = { READY_NOW, SANITY_CHECK,
 				 0x00, 0x00, 0x00 };
 	struct net_buf *buf = NULL;
-	struct bt_hci_cmd_hdr cmd_hdr;
-	struct bt_hci_acl_hdr acl_hdr;
+	struct bt_hci_cmd_hdr *cmd_hdr = (struct bt_hci_cmd_hdr *) &rxmsg[1];
+	struct bt_hci_acl_hdr *acl_hdr = (struct bt_hci_acl_hdr *) &rxmsg[1];
```

Comment:
Minor optimisation, but wouldn't a union make sense here? Also, remove the space between the typecast and variable name - I think that's more consistent with the rest of the Bluetooth code (even though there might not be a documented coding style rule that requires is)

Reply:
```checkpatch``` should be able to tell for this rule.

I personnally like the struct pointer approach more than the original memcpy approach as it is faster and sill makes clear sense of how the rxmsg buffer is interpreted. 

Could you be more clear about how the union change would improve this already optimized version ?

Reply:
I think I understand where you're getting at:
```c
union hdr_ptr {
   struct bt_hci_cmd_hdr *cmd_hdr;
   struct bt_hci_acl_hdr *acl_hdr;
};
```
(Syntax may be wrong)
And only instanciate one ```union hdr_ptr``` and pick either cmd_hdr or acl_hdr based on the context. 

We only save 1 pointer right ?

Reply:
Correct. Regarding global coding style and checkpatch, that can be discussed separately, but having a consistent style across a subsystem (whatever it might be) is something I hope is not controversial to anyone.